p2p/src/peer_manager/peerdb/address_tables/table.rs

```diff
@@ -304,6 +303,24 @@ pub mod test_utils {
         result
     }
 
+    pub fn remove_collisions(
+        table: &Table,
+        addresses: impl Iterator<Item = SocketAddress>,
+    ) -> impl Iterator<Item = SocketAddress> {
```

Comment:
Does this actually `remove` anything? All I see is a map being filled with stuff and returned. The `table` argument is a shared reference so presumably not being changed.

Reply:
Well, it produces a new iterator over addresses, where all addresses map to distinct "bucket coordinates". I couldn't think of a better name.

Reply:
Ok, I see now. Sounds like `filter` or `dedup` but not sure how to incorporate it into a name. Something like `deduplicate_by_bucket_coords` seems a bit wordy. Not a big deal anyway.

Reply:
Would `filter_out_collisions` be a better name?

Reply:
> Would `filter_out_collisions` be a better name?

Yea sounds good